dsp_interview_transcripts/pipeline/prep_output_tables.py

```diff
@@ -0,0 +1,160 @@
+OUTPUT_PATH_SUMMARY = OUTPUT_DIR / "summary_info.csv"
+
+# Create the output directory if it doesn't exist
+OUTPUT_DIR.mkdir(parents=True, exist_ok=True)
+
+opacity_condition = alt.condition(alt.datum.Name == "None", alt.value(0.1), alt.value(0.6))
+
+
+def create_scatterplot(
+    data_viz: pd.DataFrame,
+    color: str = "Name:N",
+    tooltip: List[str] = ["Name:N", "Description:N", "question:N", "text_clean:N"],
+    domain: Optional[List[Union[str, int]]] = None,
+    range_: Optional[List[str]] = None,
+) -> alt.Chart:
+    """
+    Display texts as a scatterplot.
+
+    domain and range_ are used for specifying a 3-way colour scale when the
+    texts should be coloured by sentiment.
+
+    Args:
+        data_viz (pd.DataFrame): The DataFrame containing data to visualize. Must include columns for x and y coordinates,
+            along with fields specified in `color` and `tooltip`.
+        color (str, optional): Encoding specification for the color channel. Defaults to "Name:N".
+        tooltip (List[str], optional): List of fields to display as tooltips. Defaults to ["Name:N", "Description:N", "question:N", "text_clean:N"].
+        domain (Optional[List[Union[str, int]]], optional): Custom domain values for the color scale, defining specific categories.
+            Defaults to None.
+        range_ (Optional[List[str]], optional): Custom color range for the color scale, corresponding to the domain values.
+            Defaults to None.
+
+    Returns:
+        alt.Chart: An Altair chart object representing the scatterplot, with specified color, tooltips, and interactivity.
+    """
+
+    if domain is not None and range_ is not None:
+        color = alt.Color(color, scale=alt.Scale(domain=domain, range=range_))
+    else:
+        color = alt.Color(color)
+
+    fig = (
+        alt.Chart(data_viz)
+        .mark_circle(size=50)
+        .encode(
+            x=alt.X(
+                "x:Q",
+                axis=alt.Axis(ticks=False, labels=False, title=None, grid=False),
+            ),
+            y=alt.Y(
+                "y:Q",
+                axis=alt.Axis(ticks=False, labels=False, title=None, grid=False),
+            ),
+            color=color,
+            opacity=opacity_condition,  # Ensure opacity_condition is defined elsewhere
+            tooltip=tooltip,
+        )
+        .properties(width=900, height=600)
+        .interactive()
+    )
+
+    return fig
+
+
+if __name__ == "__main__":
+    rep_docs = pd.read_csv(PROJECT_DIR / "outputs/user_messages_min_len_9_w_sentiment_topics_representative_docs.csv")
+    data = pd.read_csv(PROJECT_DIR / "outputs/user_messages_min_len_9_w_sentiment_topics.csv")
+    data_w_names = pd.read_csv(
+        PROJECT_DIR / "outputs/user_messages_min_len_9_w_sentiment_topics_with_names_descriptions.csv"
+    )
+
+    topic_counts = pd.DataFrame(data["Cluster"].value_counts()).reset_index()
+    topic_counts = topic_counts.rename(columns={"count": "N responses in topic"})
+
```

Comment:
Minor comment, but I would like us to use pandas chaining as much as possible as it makes it easier to read the code. 

```python
    topic_counts = (
        pd.DataFrame(data["Cluster"].value_counts())
        .reset_index()
        .rename(columns={"count": "N responses in topic"})
    )
```